src/Rpde/RpdeBody.php

```diff
@@ -0,0 +1,377 @@
+<?php
+
+namespace OpenActive\Rpde;
+
+use OpenActive\BaseModel;
+use OpenActive\Contracts\SerializerInterface;
+use OpenActive\Contracts\TypeCheckerInterface;
+use OpenActive\Helpers\Str;
+use OpenActive\Helpers\JsonLd as JsonLdHelper;
+use OpenActive\Concerns\TypeChecker;
+
+class RpdeBody implements SerializerInterface, TypeCheckerInterface
+{
+    use TypeChecker;
+
+    /**
+     * The default value for the license.
+     *
+     * @var string
+     */
+    protected static $default_license = "https://creativecommons.org/licenses/by/4.0/";
+
+    /**
+     * @var string
+     */
+    protected $next;
+
+    /**
+     * @var \OpenActive\Rpde\RpdeItem[]
+     */
+    protected $items;
+
+    /**
+     * @var string
+     */
+    protected $license;
+
+    /**
+     * Create a new RPDE body instance.
+     *
+     * @param array $data
+     * @return void
+     */
+    private function __construct($data)
+    {
+        foreach ($data as $key => $value) {
+            // Make sure setter is cased properly
+            $methodName = "set" . Str::pascal($key);
+
+            if(method_exists($this, $methodName) === TRUE) {
+                $this->$methodName($value);
+            }
+        }
+    }
+
+    /**
+     * @param string $feedBaseUrl
+     * @param string $modified
+     * @param string $id
+     * @param array $items
+     * @return self
+     * @throws \Exception If on or more of the provided arguments are not of a supported type.
+     */
+    public static function createFromModifiedId(
+        $feedBaseUrl,
+        $modified,
+        $id,
+        $items
+    )
+    {
+        // If the items provided are not valid,
+        // An exception will be thrown,
+        // We don't implement any further type checks on $items after this
+        $rpdeBody = new self([
+            "items" => $items,
+            "license" => static::$default_license,
+        ]);
+
+        // If there is at least one item, run validation on items array
+        if (count($items) > 0) {
+            $firstItem = $items[0];
+
+            // Checks that the afterId and afterTimestamp provided are not the
+            // first item in the feed (helps detect whether query is correct)
+            if (
+                $firstItem->getModified() === $modified &&
+                $firstItem->getId() === $id
+            ) {
+                throw new \Exception(
```

Comment:
Would it be worth having different exception classes for each type of exception?

Reply:
This has mostly been done, could you have a look at [this PR](https://github.com/openactive/models-lib/pull/14) on the models-lib repo please?
This has mostly to do with type errors in serialization.